OpenRA.Mods.Common/Traits/Attack/AttackFrontal.cs

```diff
@@ -18,7 +18,13 @@ namespace OpenRA.Mods.Common.Traits
 	[Desc("Unit got to face the target")]
 	public class AttackFrontalInfo : AttackBaseInfo, Requires<IFacingInfo>
 	{
-		public readonly int FacingTolerance = 1;
+		public readonly int FacingTolerance = 0;
+
+		public override void RulesetLoaded(Ruleset rules, ActorInfo ai)
+		{
+			if (FacingTolerance < 0 || FacingTolerance > 128)
+				throw new YamlException("Facing tolerance must be in range of [0, 128], 128 covers 360 degrees.");
```

Comment:
I believe all cases (including this PR) applies Tolerance to both sides, so that a value of 1 allows 1 facing unit be tolerated to the left and 1 facing unit tolerated to the right.